docarray/typing/tensor/ndarray.py

```diff
@@ -112,10 +112,13 @@ def validate(
         field: 'ModelField',
         config: 'BaseConfig',
     ) -> T:
+        torch = import_library('torch', raise_error=False)
         if isinstance(value, np.ndarray):
             return cls._docarray_from_native(value)
         elif isinstance(value, NdArray):
             return cast(T, value)
+        elif torch and isinstance(value, torch.Tensor):
+            return cast(T, value)
```

Comment:
I think this can be a problem with or AnyTensor and how Pydantic deal with Union. cc @JohannesMessner .

Basically if someone define a BaseDoc with a AnyTensor and pass a torch.Tensor as input it will always be converted to a NdArray

Reply:
can you share a code snippet showing when this is problematic ?